ipatests/test_integration/test_ipa_idrange_fix.py

```diff
@@ -0,0 +1,151 @@
+
+        expected_under1000 = "No IDs under 1000 found"
+        expected_nochanges = "No changes proposed for existing ranges"
+        expected_newrange = "No new ranges proposed"
+        expected_noissues = "No changes proposed, nothing to do."
+        assert expected_under1000 in result.stderr_text
+        assert expected_nochanges in result.stderr_text
+        assert expected_newrange in result.stderr_text
+        assert expected_noissues in result.stderr_text
+
+    def test_idrange_no_rid_bases(self):
+        """Test ipa-idrange-fix command with IDrange with no RID bases."""
+        self.master.run_command([
+            "ipa",
+            "idrange-add",
+            "idrange_no_rid_bases",
+            "--base-id", '10000',
+            "--range-size", '20000',
+        ])
+
+        result = self.master.run_command(["ipa-idrange-fix", "--unattended"])
+        expected_text = "RID bases updated for range 'idrange_no_rid_bases'"
+        assert expected_text in result.stderr_text
+
+        # Remove IDrange with no rid bases
+        self.master.run_command(["ipa", "idrange-del", "idrange_no_rid_bases"])
+
+    def test_users_outofrange(self):
+        """Test ipa-idrange-fix command with users out of range."""
+        for i in range(1, 20):
+            self.master.run_command([
+                "ipa",
+                "user-add",
+                "testuser{}".format(i),
+                "--first", "Test",
+                "--last", "User {}".format(i),
+                "--uid", str(100000 + i * 10),
+            ])
+
+        result = self.master.run_command(["ipa-idrange-fix", "--unattended"])
+        expected_text = r"Range '[\w\.]+_id_range_\d{3}' created successfully"
+        match = re.search(expected_text, result.stderr_text)
+        assert match is not None
+
+        # Remove users out of range and created IDrange
+        for i in range(1, 20):
+            self.master.run_command([
+                "ipa",
+                "user-del",
+                "testuser{}".format(i)
+            ])
+        self.master.run_command([
+            "ipa",
+            "idrange-del",
+            match.group(0).split(" ")[1].replace("'", "")
+        ])
+
+    def test_user_outlier(self):
+        """Test ipa-idrange-fix command with outlier user."""
+        self.master.run_command([
+            "ipa",
+            "user-add",
+            "testuser_outlier",
+            "--first", "Outlier",
+            "--last", "User",
+            "--uid", '500000',
+        ])
+
+        result = self.master.run_command(["ipa-idrange-fix", "--unattended"])
+        expected_text = "Identities that don't fit the criteria to get a new \
+range found!"
+        expected_user = "user 'Outlier User', uid=500000"
+        assert expected_text in result.stderr_text
+        assert expected_user in result.stderr_text
+
+        # Remove outlier user
+        self.master.run_command(["ipa", "user-del", "testuser_outlier"])
```

Comment:
If you remove the delete prior to the asserts then the user will be deleted even if the test fails.

Reply:
fixed in all tests